src/zero_copy/mod.rs

```diff
@@ -423,6 +476,33 @@ pub trait Parser<'a, I: Input + ?Sized, O, E: ParserExtra<'a, I> = extra::Defaul
     #[doc(hidden)]
     fn go_check(&self, inp: &mut InputRef<'a, '_, I, E>) -> PResult<Check, O, E::Error>;
 
+    /// Parse a stream with the provided configured values. This can be used to control a parser's
+    /// behavior at parse-time.
+    fn go_cfg<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E>, cfg: Self::Config) -> PResult<M, O, E::Error>
+    where
+        Self: Sized,
+    {
+        #![allow(unused_variables)]
+        self.go::<M>(inp)
+    }
+
+    #[doc(hidden)]
+    fn go_emit_cfg(&self, inp: &mut InputRef<'a, '_, I, E>, cfg: Self::Config) -> PResult<Emit, O, E::Error>;
+    #[doc(hidden)]
+    fn go_check_cfg(&self, inp: &mut InputRef<'a, '_, I, E>, cfg: Self::Config) -> PResult<Check, O, E::Error>;
+
+    /// TODO
+    fn configure<F>(self, cfg: F) -> Configure<Self, F>
+    where
+        Self: Sized,
+        F: Fn(Self::Config, &E::Context) -> Self::Config,
```

Comment:
Perhaps `Fn(&mut Self::Config, &E::Context)`? LLVM sometimes has trouble optimising moves into references, even when inlining, and I don't see the value of passing by value really, especially if the config is unsized or something.